gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -131,11 +138,29 @@ gst_tensor_query_socket_receive (GSocket * socket, GCancellable * cancellable,
   }
 
   if (avail > 0) {
+    nns_logi ("available data: %d", (gint) avail);
     read = MIN (avail, MAX_READ_SIZE);
-    outbuf = gst_buffer_new_and_alloc (read);
-    gst_buffer_map (outbuf, &map, GST_MAP_READWRITE);
+    need_alloc = (gst_buffer_get_size (outbuf) == 0);
+
+    if (need_alloc) {
+      out_mem = gst_allocator_alloc (NULL, read, NULL);
+    } else {
+      if (gst_buffer_get_size (outbuf) < read) {
+        gst_buffer_set_size (outbuf, read);
+      }
+      out_mem = gst_buffer_get_all_memory (outbuf);
+    }
+
+    nns_logi ("allocated buffer size: %lu", gst_buffer_get_size (outbuf));
```

Comment:
I think this log may not provide useful info. when `need_alloc` is true.

Reply:
Right. Moved it to log new allocated